src/Evolution/Systems/GrMhd/ValenciaDivClean/KastaunEtAl.hpp

```diff
@@ -0,0 +1,75 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+#include <limits>
+#include <optional>
+#include <string>
+
+/// \cond
+namespace EquationsOfState {
+template <bool, size_t>
+class EquationOfState;
+}  // namespace EquationsOfState
+/// \endcond
+
+namespace grmhd {
+namespace ValenciaDivClean {
+namespace PrimitiveRecoverySchemes {
+
+/// \cond
+struct PrimitiveRecoveryData;
+/// \endcond
+
+/*!
+ * \brief Compute the primitive variables from the conservative variables using
+ * the scheme of \cite Kastaun2020uxr.
+ *
+ * In the notation of the Kastaun paper, `total_energy_density` is \f$D
+ * (1+q)\f$, `momentum_density_squared` is \f$r^2 D^2\f$,
+ * `momentum_density_dot_magnetic_field` is \f$t D^{\frac{3}{2}}\f$,
+ * `magnetic_field_squared` is \f$s D\f$, and
+ * `rest_mass_density_times_lorentz_factor` is \f$D\f$.
+ * Furthermore, the returned `PrimitiveRecoveryData.rho_h_w_squared` is \f$x
+ * D\f$.  Note also that \f$h\f$ in the Palenzuela paper is the specific
```

Comment:
why the note about the palenzuela paper?

Reply:
copy-paste malfunction